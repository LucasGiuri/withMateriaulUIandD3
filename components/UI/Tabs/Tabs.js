import PropTypes from 'prop-types';
import { TabsContainer, TabContainer } from './Tabs.styles';
import Title from '../Title/Title';

const Tabs = ({setTab, tabSelected}) => {
  const selectTab = (id) => {
    setTab(id)
  };

  const titles = [
    {id: 1, title: 'Products'},
    {id: 2, title: 'Categories'}
  ];

  return (
    <TabsContainer>
      {titles.map(data => (
        <TabContainer
          tabSelected={tabSelected} 
          id={data.id}
          key={data.id} 
          onClick={() => selectTab(data.id)
        }>
          <Title key={data.id} centered={true} text={data.title} />
        </TabContainer>
      ))};
    </TabsContainer>
  )
};

Tabs.propTypes = {
  tabSelected: PropTypes.number,
  setTab: PropTypes.func
};

Tabs.defaultProps = {
  tabSelected: 1,
  setTab: () => {}
};

export default Tabs;