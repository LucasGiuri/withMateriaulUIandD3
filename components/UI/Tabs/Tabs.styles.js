import styled from 'styled-components';
import { SECONDARY, WHITE } from '../../../public/styles/colors';

export const TabsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  background: ${WHITE};
  color: ${SECONDARY};
`;

export const TabContainer = styled.div.attrs(props => ({
    isSelected: props.tabSelected === props.id,
  }))`
  display: flex;
  flex-basis: 100%;
  justify-content: center;
  tex-align: center;
  border-bottom: 1px solid ${SECONDARY};
  background-color: ${props => props.isSelected ? WHITE : SECONDARY};
  cursor: pointer;

  p {
    color: ${props => props.isSelected ? SECONDARY : WHITE};
  }
`;

