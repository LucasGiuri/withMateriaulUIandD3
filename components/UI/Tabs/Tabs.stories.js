import React from 'react';
import { storiesOf } from '@storybook/react';
import Tabs from './Tabs';

storiesOf('Tabs')
  .add('Tabs default', () => (
    <Tabs  />
  ))

  .add('Tabs second tab is selected', () => (
    <Tabs tabSelected={2} />
  ))