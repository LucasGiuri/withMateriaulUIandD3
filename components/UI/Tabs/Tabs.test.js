import renderer from 'react-test-renderer';
import Tabs from './Tabs';
import 'jest-styled-components';

describe('Tabs', () => {
  it('Tabs basic render snapshoot', () => {
    const component = renderer.create(<Tabs />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Tabs click in another tab', () => {
    const setTab = jest.fn();
    const component = (<Tabs setTab={setTab} />);
    const tree = renderer.create(component);
    const tab = tree.root.findByType(div);
    tab.props.onClick();
    expect(setTab).toHaveBeenCalledTimes(1);
  })
})
