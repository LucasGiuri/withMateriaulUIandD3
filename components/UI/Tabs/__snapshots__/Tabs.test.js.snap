// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Tabs basic render snapshoot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #FFFFFF;
  color: #3f51b5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  tex-align: center;
  border-bottom: 1px solid #3f51b5;
  background-color: #FFFFFF;
  cursor: pointer;
}

.c1 p {
  color: #3f51b5;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  tex-align: center;
  border-bottom: 1px solid #3f51b5;
  background-color: #3f51b5;
  cursor: pointer;
}

.c3 p {
  color: #FFFFFF;
}

.c2 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #3f51b5;
  font-size: 22px;
  text-align: center;
  font-weight: normal;
}

<div
  className="c0"
>
  <div
    className="c1"
    id={1}
    onClick={[Function]}
  >
    <p
      className="c2"
    >
      Products
    </p>
  </div>
  <div
    className="c3"
    id={2}
    onClick={[Function]}
  >
    <p
      className="c2"
    >
      Categories
    </p>
  </div>
  ;
</div>
`;
